Core/src/org/sleuthkit/autopsy/datasourcesummary/ui/DataSourceSummaryTabbedPane.java

```diff
@@ -33,25 +33,85 @@
  * IngestJobInfoPanel.
  */
 @Messages({
-    "DataSourceSummaryTabbedPane_countsTab_title=Counts",
+    "DataSourceSummaryTabbedPane_typesTab_title=Types",
     "DataSourceSummaryTabbedPane_detailsTab_title=Container",
     "DataSourceSummaryTabbedPane_userActivityTab_title=User Activity",
     "DataSourceSummaryTabbedPane_ingestHistoryTab_title=Ingest History",
     "DataSourceSummaryTabbedPane_recentFileTab_title=Recent Files"
 })
 public class DataSourceSummaryTabbedPane extends JTabbedPane {
 
-    private static final long serialVersionUID = 1L;
+    /**
+     * Records of tab information (i.e. title, component, function to call on
+     * new data source).
+     */
+    private static class DataSourceTab {
+
+        private final String tabTitle;
+        private final Component component;
+        private final Consumer<DataSource> onDataSource;
+
+        /**
+         * Main constructor.
+         *
+         * @param tabTitle     The title of the tab.
+         * @param component    The component to be displayed.
+         * @param onDataSource The function to be called on a new data source.
+         */
+        DataSourceTab(String tabTitle, Component component, Consumer<DataSource> onDataSource) {
+            this.tabTitle = tabTitle;
+            this.component = component;
+            this.onDataSource = onDataSource;
+        }
+
+        /**
+         * Main constructor.
+         *
+         * @param tabTitle  The title of the tab.
+         * @param component The component to be displayed.
+         *                  BaseDataSourceSummaryPanel.setDataSource method is
+         *                  used on new data source.
```

Comment:
What is this comment talking about?

Reply:
A `BaseDataSourceSummaryPanel` has a method 'setDataSource' that should be called when a new data source should be displayed in the tabbed pane.  I can rephrase.